salt/kubernetes-minion/kubelet.jinja

```diff
@@ -12,8 +12,8 @@ KUBELET_ADDRESS="--address=0.0.0.0"
 # The port for the info server to serve on
 KUBELET_PORT="--port=10250"
 
-# You may leave this blank to use the actual hostname
-KUBELET_HOSTNAME=""
+# Use machine ID as hostname to ensure uniqueness
+KUBELET_HOSTNAME="--hostname-override={{ grains['id'] }}"
```

Comment:
can we make it `{{ grains['id'] }}.{{ pillar['internal_infra_domain'] }}` ? This means if we ever decide to have a DNS server serve this info, we can easily route traffic based on the domain.